src/main/scala/no/ndla/draftapi/service/WriteService.scala

```diff
@@ -118,19 +118,33 @@ trait WriteService {
       }
     }
 
-    def publishArticle(id: Long): Try[domain.Article] = {
+    def publishArticle(id: Long): Either[FailedArticlePublish, Long] = {
```

Comment:
Hva er grunnen til at du har endret til `Either`? Misliker å bruke den syntaksen til å sende prosessinformasjon såpass at jeg er nær å kalle det et antipattern. Fordi du som kaller må vite hva som ligger i `Left` og `Right` (som jo ikke er intuitivt ut ifra navngiving, og som såvidt jeg vet kun finnes i Scala).

Reply:
Grunnen til at jeg har brukt `Either` over `Try` her er fordi det gjør hele kodeflyten enklere med tanke på hva api endepunktene skal returnere, og at `publishArticles` kaller på `publishArticle` for hver artikkel. Jeg kunne beholdt `Try` her, og håndtert dette i `publishArticles`, men denne funksjonen returnerer en struktur som inneholder alle IDer som ble publisert, i tilegg til alle som feilet å publisere. Skulle jeg håndtert en `Try` her måtte jeg endret på exceptionene som ble brukt her til å inkludere en `val id: Long` i tilegg til en `message: String` som den har nå. Sett at det bilr gjort endringere i `publishArticle` på sikt som kaster en ny type exception, er det lett å glemme at denne også må ha et `id`-felt. Ved å brukt `Either` omgår vi dette problemet da vi sier at det alltid returneres én av to mulige klasser-objekter. Det er også en konvensjon i Scala (og i Haskell tror jeg) at, når `Either` brukes til error håndtering, er `Left` error verdien og `Right` den korrekte verdien

Reply:
Mm. Ser poenget ditt. Mitt poeng er bare at dette er en så "avansert" konstruksjon at den ikke er mye brukt, noe som gjør det vanskeligere for nestemann som skal inn i denne koden å skjønne hva som egentlig foregår... (Dette er såvidt jeg vet noe av grunnen til at `Try`-konstruksjonen ble innført som er enklere for de fleste å forholde seg til). Men hvis du mener at den er nødvendig, så har jeg ikke veto-rett :)

Reply:
Jepp, bruker som regel `Try` over `Either`. Funderte litt ekstra på det, og det er en måte jeg kan skrive det om på som gjør at vi slipper `Either`. Jeg gjør et forsøk ;-)

Reply:
Jeg likte omskrivingen mye bedre :)